src/arrayMethodSort.js

```diff
@@ -5,7 +5,28 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    for (let i = 0; i < this.length; i++) {
+      for (let o = i + 1; o < this.length; o++) {
+        const mainItem = this[i];
+        const checkItem = this[o];
+        let doChange;
+
+        if (typeof compareFunction === 'function') {
+          doChange = compareFunction(mainItem, checkItem);
+        } else {
+          doChange = `${mainItem}` > `${checkItem}`;
+        }
```

Comment:
It's better to set default value for compare function. It will make  a code shorter